lib/spack/spack/solver/concretize.lp

```diff
@@ -1570,6 +1570,35 @@ compiler_used_as_a_library(node(X, Child), Hash) :-
     attr("virtual_on_edge", PackageNode, CompilerNode2, "cxx"),
     CompilerNode1 != CompilerNode2.
 
+% If compiler mixing is off for a language, each package that uses a
+% compiler within the root unification set "registers" that compiler
+% (and the root set is prohibited from having >= 2 compilers for that
+% language).
+unification_set_compiler("root", CompilerName, CompilerVersion, Language) :-
+    attr("node", node(NodeID, NodeName)),
+    attr("virtual_on_edge", node(NodeID, NodeName), node(CompilerID, CompilerName), Language),
+    attr("version", node(CompilerID, CompilerName), CompilerVersion),
+    compiler_mixing(0, Language),
+    language(Language),
+    not allow_mixing(NodeName),
```

Comment:
This needs a `#defined` to avoid:

<img width="1849" height="635" alt="Image" src="https://github.com/user-attachments/assets/8de20629-5462-4e15-9040-23297561f8c7" />